syntax = "proto2";

package protocol;

message Request {
  oneof request {
    CalibrateReq calibrate = 1;
    RelativeMoveReq relative_move = 2;
    AbsoluteMoveReq absolute_move = 3;
  }
}

message CalibrateReq {
  required bool direction = 1;
  required uint32 limit = 2;
}

message RelativeMoveReq {
  required int32 increment = 1;
}

message AbsoluteMoveReq {
  required uint32 position = 1;
}

message Response {
  oneof response {
    CalibrateResp calibrate = 1;
    RelativeMoveResp relative_move = 2;
    AbsoluteMoveResp absolute_move = 3;
  }
}

message CalibrateResp {
  required bool success = 1;
}

message RelativeMoveResp {
  required uint32 position = 1;
}

message AbsoluteMoveResp {
}
